packages/compass-schema/src/components/export-schema/export-schema-modal.tsx

```diff
@@ -0,0 +1,321 @@
+            onChange={(event) => setAdditionalProperties(event.target.checked)}
+            checked={additionalProperties}
+          />
+          <Checkbox
+            label="Require mandatory properties"
+            onChange={(event) =>
+              setRequireMandatoryProperties(event.target.checked)
+            }
+            checked={requireMandatoryProperties}
+          />
+          <Select
+            label="EJSON Format"
+            data-testid="export-schema-type-select"
+            allowDeselect={false}
+            value={relaxed ? 'relaxed' : 'canonical'}
+            aria-label="Select an EJSON format"
+            onChange={(newVal: string) => {
+              setRelaxed(newVal === 'relaxed');
+            }}
+          >
+            <Option value="relaxed">Relaxed</Option>
+            <Option value="canonical">Canonical</Option>
+          </Select>
+        </>
+      }
+    />
+  );
+};
+
+const CompassInternalSchema: React.FunctionComponent<{ schema: Schema }> = ({
+  schema,
+}) => {
+  const [maxValues, setMaxValues] = useState<number>(3);
+
+  const exportInfo: Export = useSchemaConversion(
+    schema,
+    { maxValues },
+    (schema, settings) =>
+      JSON.stringify(exportCompassInternalSchema(schema, settings), null, 2)
+  );
+
+  return (
+    <ExportPanel
+      exportInfo={exportInfo}
+      settings={
+        <>
+          <TextInput
+            label="Max sample values"
+            type="number"
+            min={0}
+            max={1000}
+            onChange={(event) => setMaxValues(+event.target.value)}
+            value={String(maxValues)}
+          />
+        </>
+      }
+    />
+  );
+};
+
+const exportSchemaModalGridStyles = css({
+  display: 'grid',
+  gridTemplateRows: 'auto 1fr',
+  gap: spacing[2],
+  height: '100%',
+});
+
+type ExportFormat = 'mongodbJsonSchema' | 'internalSchema' | 'jsonSchema';
+
+const ExportSchemaModal: React.FunctionComponent<ExportSchemaModalProps> = ({
+  schema,
+  closeExportSchema,
+  exportSchemaOpened,
+}) => {
+  const [schemaType, setSchemaType] =
+    useState<ExportFormat>('mongodbJsonSchema');
+
+  return (
+    <InfoModal
+      data-testid="export-schema-modal"
+      onClose={closeExportSchema}
+      title="Export Schema"
+      size="large"
+      open={exportSchemaOpened}
+    >
+      <div className={exportSchemaModalGridStyles}>
+        <Select
+          label="Format"
+          data-testid="export-schema-type-select"
+          allowDeselect={false}
+          value={schemaType}
+          aria-label="Select an export format"
+          onChange={(newVal: string) => {
+            setSchemaType(newVal as ExportFormat);
+          }}
+        >
+          <Option value="mongodbJsonSchema">
+            MongoDB Validation Schema ($jsonSchema)
+          </Option>
+          <Option value="jsonSchema">JSON Schema (EJSON)</Option>
+          <Option value="internalSchema">Compass Internal Schema</Option>
```

Comment:
We're providing this for compatibility with the legacy feature that used this format until now? "Internal" sounds a bit like users aren't really supposed to rely on it :)

Reply:
good point yeah, not sure what we could call that instead, or if we want to just drop it, maybe we can add an "x-samples: [...]" to the json schema and just get rid of this one